.github/workflows/ci.yaml

```diff
@@ -4,26 +4,37 @@ on:
   - push
   - pull_request
 jobs:
+  pyvers:
+    name: Enumerate supported Python versions
+    runs-on: ubuntu-latest
+    outputs:
+      versions: ${{ steps.enumerator.outputs.versions }}
+    steps:
+      - id: enumerator
+        run: |
+          echo ::set-output "name=versions::['3.7','3.8','3.9','3.10']"
   lint:
     name: Run linters
     runs-on: ubuntu-latest
+    needs: pyvers
     steps:
       - name: Checkout source code
         uses: actions/checkout@v2
-      - name: Set up Python 3.9
+      - name: Set up Python 3
         uses: actions/setup-python@v2
         with:
-          python-version: 3.9
+          python-version: ${{ fromJson(needs.pyvers.outputs.versions)[-1] }}
```

Comment:
I think this is a little too clever :see_no_evil: Personally, I'd rather have a little duplication and make a little easier to grok. Could we stick to a static list as before?

Reply:
Sorry, I meant I would rather just update these lines to use `"3.10"` instead of `3.9`, and then to update the below to use `["3.7", "3.8", "3.9", "3.10"]`.

Reply:
Oops! This appears after I pushed another revision.

The reason I start this short patch is that I found you have a commit that updates `tox.ini` and other documents, but missed these unit tests. I would recommend that we try to generate supported versions in github actions yml, so that it helps to validate the correctness of such version bumps. It would be even better in my opinion if some how the CI gives a warning if a new python version is available. Does this sound good to you?